lib/matplotlib/__init__.py

```diff
@@ -356,64 +469,39 @@ def checkdep_inkscape():
 def checkdep_ps_distiller(s):
     if not s:
         return False
-
-    flag = True
-    gs_exec, gs_v = checkdep_ghostscript()
-    if not gs_exec:
-        flag = False
-        _log.warning('matplotlibrc ps.usedistiller option can not be used '
-                     'unless ghostscript 9.0 or later is installed on your '
-                     'system.')
-
-    if s == 'xpdf':
-        pdftops_req = '3.0'
-        pdftops_req_alt = '0.9'  # poppler version numbers, ugh
-        pdftops_v = checkdep_pdftops()
-        if compare_versions(pdftops_v, pdftops_req):
-            pass
-        elif (compare_versions(pdftops_v, pdftops_req_alt) and not
-              compare_versions(pdftops_v, '1.0')):
-            pass
-        else:
-            flag = False
-            _log.warning('matplotlibrc ps.usedistiller can not be set to xpdf '
-                         'unless xpdf-%s or later is installed on your '
-                         'system.', pdftops_req)
-
-    if flag:
-        return s
-    else:
+    try:
+        get_executable_info("gs")
+    except FileNotFoundError:
+        _log.warning(
+            "Setting rcParams['ps.usedistiller'] requires ghostscript.")
         return False
+    if s == "xpdf":
+        try:
+            get_executable_info("pdftops")
+        except FileNotFoundError:
+            _log.warning(
+                "Setting rcParams['ps.usedistiller'] to 'xpdf' requires xpdf.")
+            return False
+    return s
 
 
 def checkdep_usetex(s):
     if not s:
         return False
-
-    gs_req = '9.00'
-    dvipng_req = '1.6'
-    flag = True
-
-    if shutil.which("tex") is None:
-        flag = False
-        _log.warning('matplotlibrc text.usetex option can not be used unless '
-                     'TeX is installed on your system.')
-
-    dvipng_v = checkdep_dvipng()
-    if not compare_versions(dvipng_v, dvipng_req):
-        flag = False
-        _log.warning('matplotlibrc text.usetex can not be used with *Agg '
-                     'backend unless dvipng-%s or later is installed on '
-                     'your system.', dvipng_req)
-
-    gs_exec, gs_v = checkdep_ghostscript()
-    if not compare_versions(gs_v, gs_req):
-        flag = False
-        _log.warning('matplotlibrc text.usetex can not be used unless '
-                     'ghostscript-%s or later is installed on your system.',
-                     gs_req)
-
-    return flag
+    if not shutil.which("tex"):
+        _log.warning("usetex mode requires TeX.")
+        return False
+    try:
```

Comment:
But this is actually only used as a test marker (for non-test use, we don't run checkdep_usetex and instead just... do whatever is explained in https://github.com/matplotlib/matplotlib/pull/13285, which you already approved :)), so that doesn't really matter.

Reply:
:+1: